### PR TITLE
QVariant add primitives support

### PR DESCRIPTION
**Pull request by ahayzen-kdab**
Related to #11

### CODE REVIEW

**Review by LeonMatthesKDAB — COMMENTED** [review]
One minor nitpick on a switch statement that generates a lot of warnings on my machine.
Also wondering why u64/i64 support is missing?

Otherwise looks good :+1:

File: cxx-qt-gen/src/cxx_qt.cpp [hunk 1]
```diff
@@ -253,9 +253,16 @@ static_assert(sizeof(QVariant) <= sizeof(std::size_t[2]),
...
+  Bool = 1,
+  F32 = 2,
+  F64 = 3,
+  I8 = 4,
+  I16 = 5,
+  I32 = 6,
+  String = 7,
+  U8 = 8,
+  U16 = 9,
+  U32 = 10,
```
[the remark this answers is not in the document]
Comment (ahayzen-kdab) [reply cxx-qt-gen/src/cxx_qt.cpp:265]: Yes, we don't support u64 or i64 yet https://github.com/KDAB/cxx-qt/issues/36 this is due to the unfortunate situation of qint64 and std::int64_t  disagreeing on the type definition :'(

File: cxx-qt-gen/src/cxx_qt.cpp [hunk 2]
```diff
@@ -276,44 +313,110 @@ extern "C"
...
+
+  void cxxqt1$qvariant$init$from$u32(QVariant* self, quint32 i) noexcept
+  {
+    new (self) QVariant();
+    *self = QVariant::fromValue(i);
   }
 
   QVariantType cxxqt1$qvariant$get$type(const QVariant& self) noexcept
   {
     switch (self.type()) {
```
[the remark this answers is not in the document]
Comment (ahayzen-kdab) [reply cxx-qt-gen/src/cxx_qt.cpp]: Right, used a static_cast to ensure type is correct :-)

File: cxx-qt-gen/src/cxx_qt.cpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (LeonMatthesKDAB) [reply cxx-qt-gen/src/cxx_qt.cpp:265]: Ah, okay, then that's something to do later :+1:

**Review by LeonMatthesKDAB — APPROVED** [review]
LGTM

**Pull request merged**